arch/x86/Makefile.in

```diff
@@ -7,45 +7,50 @@ CFLAGS=
 SFLAGS=
 INCLUDES=
 
-SRCDIR=
-SRCTOP=
+SRCDIR=.
+SRCTOP=../..
 
 all: x86.o x86.lo fill_window_sse.o fill_window_sse.lo deflate_quick.o deflate_quick.lo insert_string_sse.o insert_string_sse.lo crc_folding.o crc_folding.lo
 
-x86.o:
+x86.o: $(SRCDIR)/x86.c
 	$(CC) $(CFLAGS) $(INCLUDES) -c -o $@ $(SRCDIR)/x86.c
 
-x86.lo:
+x86.lo: $(SRCDIR)/x86.c
 	$(CC) $(SFLAGS) $(INCLUDES) -c -o $@ $(SRCDIR)/x86.c
 
-fill_window_sse.o:
+fill_window_sse.o: $(SRCDIR)/fill_window_sse.c
 	$(CC) $(CFLAGS) -msse2 $(INCLUDES) -c -o $@ $(SRCDIR)/fill_window_sse.c
 
-fill_window_sse.lo:
+fill_window_sse.lo: $(SRCDIR)/fill_window_sse.c
 	$(CC) $(SFLAGS) -msse2 -DPIC $(INCLUDES) -c -o $@ $(SRCDIR)/fill_window_sse.c
 
-deflate_quick.o:
+deflate_quick.o: $(SRCDIR)/deflate_quick.c
 	$(CC) $(CFLAGS) -msse4 $(INCLUDES) -c -o $@ $(SRCDIR)/deflate_quick.c
 
-deflate_quick.lo:
+deflate_quick.lo: $(SRCDIR)/deflate_quick.c
 	$(CC) $(SFLAGS) -msse4 -DPIC $(INCLUDES) -c -o $@ $(SRCDIR)/deflate_quick.c
 
-insert_string_sse.o:
+insert_string_sse.o: $(SRCDIR)/insert_string_sse.c
 	$(CC) $(CFLAGS) -msse4 $(INCLUDES) -c -o $@ $(SRCDIR)/insert_string_sse.c
 
-insert_string_sse.lo:
+insert_string_sse.lo: $(SRCDIR)/insert_string_sse.c
 	$(CC) $(SFLAGS) -msse4 -DPIC $(INCLUDES) -c -o $@ $(SRCDIR)/insert_string_sse.c
 
-crc_folding.o:
+crc_folding.o: $(SRCDIR)/crc_folding.c
 	$(CC) $(CFLAGS) -mpclmul -msse4 $(INCLUDES) -c -o $@ $(SRCDIR)/crc_folding.c
 
-crc_folding.lo:
+crc_folding.lo: $(SRCDIR)/crc_folding.c
 	$(CC) $(SFLAGS) -mpclmul -msse4 -DPIC $(INCLUDES) -c -o $@ $(SRCDIR)/crc_folding.c
 
+crc_pclmulqdq.o: $(SRCDIR)/crc_pclmulqdq.c
+	$(CC) $(CFLAGS) -msse4 $(INCLUDES) -c -o $@ $(SRCDIR)/crc_pclmulqdq.c
+
+crc_pclmulqdq.lo: $(SRCDIR)/crc_pclmulqdq.c
+	$(CC) $(SFLAGS) -msse4 -DPIC $(INCLUDES) -c -o $@ $(SRCDIR)/crc_pclmulqdq.c
```

Comment:
These two should not depend on -msse4 (or -mpclmul) because it has code that checks the cpuid flags for whether the pclmul instruction sets are present. So if zlib-ng is compiled with this, but the running cpu only supports say SSE2, then the code could crash if the compiler chooses to use sse3 or sse4 instructions for the functions in this file.
Luckily none of the functions in this file actually depend on SSE4 or PCLMUL instruction sets, so it can just be compiled without any such options.